### DISCUSSION TITLE
How to connect to services installed on the VPN serwer.

### DISCUSSION THREAD

**Discussion by aleksander237** [strongswan/strongswan#1666] [Q&A] [answered]
Hello,

I followed this instruction https://www.digitalocean.com/community/tutorials/how-to-set-up-an-ikev2-vpn-server-with-strongswan-on-ubuntu-22-04 of setting up IKEv2 on ubuntu 22.04. Everything works, but let's say that I have a MariaDB server or NGINX server installed on the same machine (the one that acts as VPN Server), and I want these services to be reachable by my client only via VPN tunel. 

Let's say that my left/right config looks like this:

    ...
    left=%any
    leftid=@vpn.example.com
    leftcert=server-cert.pem
    leftsendcert=always
    leftsubnet=0.0.0.0/0

    right=%any
    rightid=%any
    rightauth=eap-mschapv2
    rightsourceip=192.168.97.11/24 - 192.168.97.50/24
    rightdns=8.8.8.8,8.8.4.4
    rightsendcert=never
    ...

I imagine that my server also needs to have an IP address from the subnet 192.168.97.0/24 (97.1 for example), but where I should set this IP? With Wire Guards Everything Was "simple" because on both sides there are interfaces which represents both ends of tunnel. How can I achieve this effect with IKEv2?

Best Regards,
Aleksander

**Comment by tobiasbrunner** [comment]
> I imagine that my server also needs to have an IP address from the subnet 192.168.97.0/24 (97.1 for example)

It does not necessarily have to be from that subnet. It could be from any other subnet (with `leftsubnet=0.0.0.0/0` client will send all traffic to the server unless they exclude some traffic locally), or even be the server's public address. However, the latter might not work with clients that don't send traffic for the VPN server's IP address via tunnel (e.g. macOS/iOS). You can use policy matching via iptables to only allow traffic to specific IPs and ports via VPN tunnel.

> but where I should set this IP?

Just assign it to any local interface (could be `lo`). Using [XFRM interfaces](https://docs.strongswan.org/docs/5.9/features/routeBasedVpn.html#_xfrm_interfaces_on_linux) is also an option if you feel that's easier.

> rightsourceip=192.168.97.11/24 - 192.168.97.50/24

That's not the correct syntax. Either use a range (`192.168.97.11 - 192.168.97.50`) or use a subnet, optionally with a specific base  address e.g. `192.168.97.11/24` (includes all addresses in `192.168.97.0/24` starting with `192.168.97.11`).

**Comment by aleksander237** [comment]
Thanks for your answer

I've created xfrm interface:
```console
ip link add ipsec0 type xfrm dev eth0 if_id 42
```
and assigned the ip address
```console
ip addr add 192.168.80.1/24 dev ipsec0
```
The nginx listens on 192.168.80.1 and everything works fine from the windows client perspective - while connected to VPN i can ping the 192.168.80.1 address and open my "private" website. However, I'm wondering about one thing. Let's say I want to ping the 192.168.97.11 (windows client) from my ubuntu server (which is also the VPN gateway), what should be configured to make it possible, how to tell linux to send packets to 192.168.97.0/24 network via tunel? Again with Wire Guard it was pretty straightforward - on linux server there was a virtual interface representing the end of the tunel. My current interface table looks like this:
```console
1: lo: <LOOPBACK,UP,LOWER_UP> mtu 65536 qdisc noqueue state UNKNOWN group default qlen 1000
    link/loopback 00:00:00:00:00:00 brd 00:00:00:00:00:00
    inet 127.0.0.1/8 scope host lo
       valid_lft forever preferred_lft forever
    inet6 ::1/128 scope host
       valid_lft forever preferred_lft forever
2: eth0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc fq_codel state UP group default qlen 1000
    link/ether XX:XX:XX:XX:XX:XX brd ff:ff:ff:ff:ff:ff
    inet XX.XX.XX.XXX/32 scope global dynamic eth0
       valid_lft 76808sec preferred_lft 76808sec
    inet6 XXXX:XXXX:XXXX:XXXX::1/64 scope global
       valid_lft forever preferred_lft forever
    inet6 XXXX::XXXX:XXX:XXXX:XXXX/64 scope link
       valid_lft forever preferred_lft forever
5: ipsec0@eth0: <NOARP> mtu 1500 qdisc noop state DOWN group default qlen 1000
    link/none
    inet 192.168.80.1/24 scope global ipsec0
       valid_lft forever preferred_lft forever

```

**Reply by tobiasbrunner** [reply, marked as answer]
> how to tell linux to send packets to 192.168.97.0/24 network via tunel?

That seems to work already. Or how else is response traffic between `192.168.80.1` and `192.168.97.11` routed from the server to the client?

In general, it depends on whether you just use the XFRM interface to install an IP address on (for which you, again, could just use an existing interface or any other type of virtual interface, doesn't have to be an XFRM interface), or you actually configured strongSwan to use it for the IPsec SAs and policies (only possible via `swanctl.conf`, so if you still use `ipsec.conf`, that's not an option). If the latter, you just configure a route via `ipsec0` interface for `192.168.97.0/24`. If the former, you don't have to do anything special as strongSwan installs routes for the virtual IPs assigned to clients (check routing table 220). The only thing you have to be aware of are possible NAT rules that map traffic from your clients to the public IP of the server (see [here](https://docs.strongswan.org/docs/5.9/howtos/forwarding.html#_general_nat_problems)).

**Reply by aleksander237** [reply]
> That seems to work already. Or how else is response traffic between 192.168.80.1 and 192.168.97.11 routed from the server to the client?

The ping was working while initialised from the client's side:
```bash
ping 192.168.80.1
```
The ping wasn't working while initialised from the server's side:
```bash
ping 192.168.97.11
```
My intention of using XFRM interface was to install the IP address for nginx. I was a bit confused about interfaces, but you clarified it to me. Thanks for your time!
